### ISSUE TITLE
Allow for type number on label of `state.emit`

### ISSUE THREAD

**Issue by stelcheck** [mage/mage#40]
It would be nice, since it would allow for the following kind of code:

```typescript
import * as mage from 'mage'

export enum events {
    Success,
    Failure,
    Dunno
}

export function emitSuccess(state: mage.core.IState) {
    state.emit('123', events.Success, 'yay')
}
```

@christopherobin @ronkorving any objections?

**Comment by ronkorving** [comment]
The Web SDK explicitly expects event names to be strings. The event manager will emit 3 events for an event named `"foo.bar.helloworld"`: 

1. `"foo.bar.helloworld"`
2. `"foo.bar"`
3. `"foo"`

Event listeners at each level receive a `path` argument that contains the full string.

So allowing numbers would change semantics quite a bit.

**However**:

While in the past that model served us well, things have changed over the years, and we currently have no code (at least not built into MAGE) that depends on that to work. In fact, I would be quite happy to see this change to simply emit the event data only and not pass the "event path". A BC breaking change, but one that I think at this point would be a good change to make.

**However 2**:

Having said all that (yes, I wanna get rid of the event path argument and cascading event emission), `EventEmitter` emits strings, and emitting numbers would make the client-side cast those integers into strings: `mage.eventManager.on(123, function () {})` would behave identical to `mage.eventManager.on('123', function () {});` which I would consider to be incorrect.

Sure we can change the EventManager to be less of an `EventEmitter` and be able to emit integers as well, but on the web I would argue this would feel quite wrong.

So let me turn this around: do you have an actual use-case? Or is this all hypothetical? I understand wanting to emit numbers for status etc, but that status would normally be contextualized. Ie, by coming back from a user command response, *or* by being contextualized by an event *name* (eg: `mage.eventManager.on('error', function (code) {})`.)

**Comment by stelcheck** [comment]
Cascading events like this would be a nightmare to deal with for strongly typed languages: whether we wish to make it go or not is, however, not a discussion we need to have now, since we can always deal with this by not using the dot notation at all.

As for the type to be emitted, I see no issues with number being casted to strings internally. All I want is to be able to input a number instead of a string.

Use-case is as specified above. Instead of using strings in place, now you can list all event names into an enum, which I feel is safer and cleaner.

**Comment by ronkorving** [comment]
@stelcheck merged in #41 , right?
